src/js/view/controls/components/timeslider.js

```diff
@@ -280,6 +280,12 @@ class TimeSlider extends Slider {
     }
 
     updateCues(model, cues) {
+        // If updateCues is called with cues === [], clear the cues on the model.
+        // This will clear out the cues on the model on new playlist items.
+        if (cues && cues.length === 0) {
+            model.set('cues', cues);
```

Comment:
This line executes this method when the model is changed, so you should never change the model in this method `.on('change:cues', this.updateCues, this)`